spring/boot2-webflux-autoconfigure/src/main/java/com/linecorp/armeria/spring/web/reactive/ArmeriaReactiveWebServerFactory.java

```diff
@@ -134,84 +133,118 @@ public WebServer getWebServer(HttpHandler httpHandler) {
-            logger.warn(
-                    "Cannot disable HTTP/2 protocol for Armeria server. It will be enabled automatically.");
+        if (armeriaSettings != null) {
+            configureServerWithArmeriaSettings(sb, armeriaSettings,
+                                               findBeans(ArmeriaServerConfigurator.class),
+                                               findBeans(Consumer.class, ServerBuilder.class),
+                                               findBeans(DocServiceConfigurator.class),
+                                               firstNonNull(findBean(MeterRegistry.class),
+                                                            Metrics.globalRegistry),
+                                               findBeans(HealthChecker.class),
+                                               findBeans(HealthCheckServiceConfigurator.class),
+                                               meterIdPrefixFunctionOrDefault());
         }
 
-        final InetAddress address = getAddress();
-        final int port = ensureValidPort(getPort());
-        if (address != null) {
-            sb.port(new InetSocketAddress(address, port), protocol);
-        } else {
-            sb.port(port, protocol);
-        }
+        // In the property file, both Spring and Armeria port configuration can coexist.
+        // We use following to define primaryAddress, primaryLocalPort and primarySessionProtocol:
+        // (we consider Spring port is not specified when it's 0.)
+        // 1) Both Armeria and Spring port were specified:
+        //    - primaryAddress = Spring address
+        //    - primaryLocalPort = Spring port
+        //    - primarySessionProtocol = depends on the springSsl.isEnabled()
+        // 2) Only Spring port was specified:
+        //    - primaryAddress = Spring address
+        //    - primaryLocalPort = Spring port
+        //    - primarySessionProtocol = depends on the springSsl.isEnabled()
+        // 3) Only Armeria port was specified:
+        //    - primaryAddress = null;
+        //    - primaryLocalPort = the port of the first Armeria Port
+        //    - primarySessionProtocol = the session protocol of the first Armeria Port
+        // 4) No port was specified:
+        //    a) The port is configured by other ways (e.g. ArmeriaServerConfigurator)
+        //       - primaryAddress = null
+        //       - primaryLocalPort = the port of the first Armeria Port
+        //       - primarySessionProtocol = the session protocol of the first Armeria Port
+        //    b) The port is not specified.
+        //       - primaryAddress = Spring address
+        //       - primaryLocalPort = Spring port
+        //       - primarySessionProtocol = depends on the armeriaSsl.isEnabled().
+        //                               If armeriaSsl is null use SpringSsl
+        //
+        // Please note that Armeria TLS configuration has precedence over Spring SSL.
```

Comment:
The current implementation is good enough. 
However, the detailed condition is a bit complicated. Would it be better to have a dedicated method and entity when we refactor code later?
```java
class PrimaryAddress {
    boolean needToBind;
    InetAddress address;
    int port;
    SessionProtocol protocol;
}

private PrimaryAddress getPrimaryAddress(ArmeriaSettings ...) { 
   ...
}
```

Reply:
That's a good suggestion. 👍 I will consider that.